test/hotspot/jtreg/compiler/loopopts/superword/TestMovingLoadBeforeStore2.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+
+/**
+ * @test
+ * @requires vm.compiler2.enabled
+ * @bug 8316594
+ * @summary In SuperWord::output, LoadVector can be moved before StoreVector, but only if it is proven to be safe.
+ * @library /test/lib
+ * @run main/othervm -XX:CompileCommand=compileonly,compiler.loopopts.superword.TestMovingLoadBeforeStore2::test
+ *                   -XX:LoopUnrollLimit=250 -Xbatch
+ *                   -XX:+UnlockDiagnosticVMOptions -XX:+StressLCM
+ *                   compiler.loopopts.superword.TestMovingLoadBeforeStore2
+ */
+
+package compiler.loopopts.superword;
+import java.util.Random;
+import jdk.test.lib.Utils;
+
+public class TestMovingLoadBeforeStore2 {
```

Comment:
I think the two tests should be merged or is there a reason to have them separate? If you want different command line arguments, you can add two `@run`. Even if they are not needed for both, I think it doesn't hurt.